geonode/base/models.py

```diff
@@ -64,7 +64,7 @@ class ContactRole(models.Model):
     """
     ContactRole is an intermediate model to bind Profiles as Contacts to Resources and apply roles.
     """
-    resource = models.ForeignKey('ResourceBase')
+    resource = models.ForeignKey('ResourceBase', blank=True, null=True)
```

Comment:
Yeps. Currently GeoNode has a hierarchical model and there is no way to allow it recognize that a Layer is "also" a resourcebase by using fixtures. Unfortunately this is a core issue of DJango and I honestly did not found any solution except relaxing the model.